test/stream_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:test/test.dart';
+import 'dart:async';
+import 'entity.dart';
+import 'test_env.dart';
+import 'objectbox.g.dart';
+import 'package:objectbox/src/observable.dart';
+
+// ignore_for_file: non_constant_identifier_names
+
+void main() {
+  TestEnv env;
+  Box box;
+
+  setUp(() {
+    env = TestEnv('streams');
+    box = env.box;
+  });
+
+  test('Subscribe to stream of entities', () async {
+
+    final result = <String>[];
+    final text = TestEntity_.tString;
+    final condition = text.notNull();
+    final query = box.query(condition).order(text).build();
+    final queryStream = query.findStream();
+    final subscription = queryStream.listen((list) {
+      final str = list.map((t) => t.tString).toList().join(', ');
+      result.add(str);
+    });
+
+    box.put(TestEntity(tString: 'Hello world'));
+    box.putMany(<TestEntity>[ TestEntity(tString: 'Goodbye'),
+      TestEntity(tString: 'for now') ]);
+
+    await Future.delayed(Duration(seconds: 0)); // ffi explodes without
+    expect(result,
+        ['for now, Goodbye, Hello world', 'for now, Goodbye, Hello world']);
```

Comment:
Shouldn't this be `['Hello world', 'for now, Goodbye, Hello world']? E.g. first put is one event, second putMany the other. Or am I missing something here?

Reply:
Doh.

Reply:
I guess for some reason the puts execute completely before the observables fire. Strange behaviour.